source/server/filter_chain_manager_impl.cc

```diff
@@ -156,20 +156,26 @@ void FilterChainManagerImpl::addFilterChains(
       filter_chains;
   uint32_t new_filter_chain_size = 0;
   for (const auto& filter_chain : filter_chain_span) {
+    std::string fc_name;
+    if (!filter_chain->name().empty()) {
+      fc_name = filter_chain->name();
+    } else {
+      fc_name = parent_context_.api().randomGenerator().uuid();
```

Comment:
How can we use this UUID to be correlated with actual config? I don't think this will be dumped into config_dump, is it?

In general I would think this should be generated from control-plane.

Reply:
In case you want to added to config_dump: 
The newly assigned name should not break listener in place update. 

If the goal is to debug, the low hanging fruit is to dump the conflicting filter chain proto message.